packages/tools/rkbin/package.mk

```diff
@@ -2,11 +2,54 @@
 # Copyright (C) 2017-present Team LibreELEC (https://libreelec.tv)
 
 PKG_NAME="rkbin"
-PKG_VERSION="b0c100f1a260d807df450019774993c761beb79d"
-PKG_SHA256="c6ebf8ab556e071e3b067540e95aecff650143f0c97e129cd40c837a4f11a881"
+PKG_VERSION="c41b714cacd249e3ef69b2bbe774da5095eefd72"
+PKG_SHA256="c5ad3002e3d7ff50edff84f8874e96d3069be8292c5febd01fe1cf557a2d0c06"
 PKG_ARCH="arm aarch64"
 PKG_LICENSE="nonfree"
 PKG_SITE="https://github.com/rockchip-linux/rkbin"
 PKG_URL="https://github.com/rockchip-linux/rkbin/archive/${PKG_VERSION}.tar.gz"
 PKG_LONGDESC="rkbin: Rockchip Firmware and Tool Binaries"
 PKG_TOOLCHAIN="manual"
+PKG_STAMP="$UBOOT_SYSTEM"
+
+make_target() {
+  PKG_SOC=$("${ROOT}/${SCRIPTS}/uboot_helper" "${PROJECT}" "${DEVICE}" "${UBOOT_SYSTEM}" soc)
+  PKG_SOC=${PKG_SOC:-${DEVICE}}
+
+  PKG_BOOT_INI="RKBOOT/${PKG_SOC}MINIALL.ini"
+  if [ ! -f "${PKG_BOOT_INI}" ]; then
+    PKG_BOOT_INI="RKBOOT/${DEVICE}MINIALL.ini"
+  fi
+  if [ -f "${PKG_BOOT_INI}" ]; then
+    PKG_FILE=$(sed -nr "/^\[LOADER_OPTION\]/ { :l /^FlashData[ ]*=/ { s/[^=]*=[ ]*//; p; q;}; n; b l;}" "${PKG_BOOT_INI}")
+    if [ -f "${PKG_FILE}" ]; then
+      cp -av "${PKG_FILE}" ddr.bin
+
+      # Override sdram frequency
```

Comment:
do you need to remove the `override` in another place?

Reply:
This override can be traced back to when rockchip used to provide a 786MHz ddr init blob for RK3328.
However, they dropped these and only supplied a 333MHz version for newer versions of ddr init blobs for RK3328.

The only difference between the 333MHz and 786MHz blobs was an entry in the embedded ddr timing table, these `sed` command basically replace `333` with `786` in the embedded ddr timing table of the ddr blob.
This is/was needed because mainline linux does not have any driver that can change ddr frequencey from initial 666MHz to 1572MHz on RK3328 when a board is using the 333MHz ddr init blob and mainline linux.

These days it is only tv-boxes boards that use `rockchip_legacy_boot` flag and is affected by this, remaining RK3328 boards do ddr init using U-Boot TPL.